docs/app/docs-infra/hooks/use-search/page.mdx

```diff
@@ -293,11 +292,16 @@ Override how pages are converted to search entries:
 const { search } = useSearch({
   sitemap: () => import('./sitemap'),
   flattenPage: (page, sectionData) => {
+    const slug = page.title
```

Comment:
Duplicate logic with https://github.com/mui/mui-public/pull/985/files#diff-19ec2058ffa4d961a1b2d87f221d2a1b5049e86d4b8e379acc0f71eedd26d1e9R27

What are the different use-cases and can we create a slug algorithm for each use case to be used by our docs?

* markdown fragment to HTML id
* ...?